src/SWP/Component/TemplatesSystem/Gimme/Context/Context.php

```diff
@@ -27,14 +30,143 @@ class Context implements \ArrayAccess
     /**
      * Array will all registered meta types.
      *
-     * @var \SWP\TemplatesSystem\Gimme\Meta\Meta[]
+     * @var Meta[]
      */
     protected $registeredMeta = [];
 
     /**
-     * Set current context page informations.
+     * Array with available meta configs.
      *
-     * @param string[] $currentPage
+     * @var array
+     */
+    protected $availableConfigs = [];
+
+    /**
+     * @var Cache
+     */
+    protected $metadataCache;
+
+    /**
+     * @var string
+     */
+    protected $configsPath;
+
+    /**
+     * Context constructor.
+     *
+     * @param Cache  $metadataCache
+     * @param string $configsPath
+     */
+    public function __construct(Cache $metadataCache, $configsPath = null)
+    {
+        $this->metadataCache = $metadataCache;
+        $this->configsPath = $configsPath;
+
+        if (null !== $configsPath) {
+            $finder = new Finder();
```

Comment:
Agree we can scan in on first `getAvailableConfigs()` call. So i will need to use it always instead just touching property (inside this class).